### PR TITLE
blog: bpfman integration with AF_XDP

### PR DESCRIPTION
**Pull request by maryamtahhan**
A blog post on bpfman integration with the AF_XDP DP and CNI

### CODE REVIEW

File: docs/blog/posts/af_xdp-dp-integration.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+## AF_XDP DP and CNI's integration with bpfman in images
+
+The following sections will present the evolution of the AF_XDP DP and CNI
+from independent bpf program management to leveraging bpfman to manage eBPF programs on
+their behalf.
+
+### AF_XDP DP and CNI managing eBPF programs independently
+
+![AF_XDP DP and CNI managing eBPF programs independently](./img/2024-02-27/af_xdp.png)
+
```
Comment (maryamtahhan) [inline docs/blog/posts/af_xdp-dp-integration.md:107]: I can call out the steps in detail here, WDYT?

Comment (astoycos) [reply docs/blog/posts/af_xdp-dp-integration.md:107]: It think that makes sense!! Most folks reading will come with much less context on what af_xdp actually is so feel free to talk more details!

**Review by astoycos — COMMENTED** [review]
Looking really good @maryamtahhan!! Thanks for opening this, mostly nits  and re-wording pretty much.

A little more about the details of AF_XDP wouldn't be bad either, at the beginning maybe try and reference a real world example of an application using AD_XDP just to hook readers!

File: docs/blog/posts/af_xdp-dp-integration.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+of a series of demos that show the migration of a DPDK application to AF_XDP (without)
+any application modification. The demo can be watched below:
+
+[<img src="https://img.youtube.com/vi/wcmSO5HwNJQ/hqdefault.jpg" width="600" height="300"
+/>](https://www.youtube.com/embed/wcmSO5HwNJQ)
+
+## AF_XDP DP and CNI's integration with bpfman in images
+
+The following sections will present the evolution of the AF_XDP DP and CNI
+from independent bpf program management to leveraging bpfman to manage eBPF programs on
```
Comment (Billy99) [inline docs/blog/posts/af_xdp-dp-integration.md:101]:
```suggestion
from independent eBPF program management to leveraging bpfman to manage eBPF programs on
```

File: docs/blog/posts/af_xdp-dp-integration.md (hunk 1 above)
Comment (Billy99) [reply docs/blog/posts/af_xdp-dp-integration.md:107]: And/Or highlight the steps where bpfman made things better, like Step 9 handshake no longer needed.

**Comment by maryamtahhan** [comment]
Still working on some updates - should be done by EOD in another push

**Comment by astoycos** [comment]
Ahh shoot I'm sorry @maryamtahhan I accidentally merged this, feel free just to push those fixes in a follow up if needed!

**Comment by maryamtahhan** [comment]
lol no worries - will fix up the bpfd bpfman diagram typo and push it in a few mins

File: docs/blog/posts/af_xdp-dp-integration.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+  - This presented a difficulty particularly when trying to find/update appropriate
+    base container images to use for the AF_XDP device plugin. Different images
+    supported different versions of eBPF management libraries (i.e libbpf or libxdp) which
+    forced multiple changes around the loading and attaching of the base eBPF program.
+
+  - Additionally the CNI runs as a binary on the Kubernetes node so we would need to
+    statically compile libbpf/libxdp as part of the CNI.
+
+  > **NOTE**: The CNI has been extended to signal the AF_XDP DP on pod termination (via gRPC).
```
[the remark this answers is not in the document]
Comment (maryamtahhan) [reply docs/blog/posts/af_xdp-dp-integration.md:130]: it was extended to do that for bpf map pinning anyway - before the bpfman integration. 
I will make it more clear

**Pull request merged**